broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/perf/LargeStateControllerPerformanceTest.java

```diff
@@ -61,18 +66,30 @@ public void setup() throws Throwable {
         VALUE_SIZE);
     context = new TestState().generateContext(KEY_COUNT, KEY_SIZE, VALUE_SIZE);
 
-    // it's fine to check the size of the root folder since it basically only contains the snapshot
-    final var snapshotSize = FileUtils.sizeOfDirectory(context.temporaryFolder().toFile());
+    final var snapshotSize =
+        StreamSupport.stream(
+                Files.fileTraverser()
+                    .breadthFirst(context.temporaryFolder().toFile())
+                    .spliterator(),
+                true)
+            .mapToLong(File::length)
+            .sum();
     LOG.info(
         "Created a test size with a total size of {}",
-        FileUtils.byteCountToDisplaySize(snapshotSize));
+        String.format("%.3f", snapshotSize / ONE_GB));
   }
 
   @TearDown
   public void tearDown() throws Exception {
     context.close();
   }
 
+  @JMHTest("measureStateRecovery")
+  void shouldRecoverStateWithinDeviation(final JMHTestCase testCase) {
+    // given - when - then
+    testCase.run().isWithinDeviation(3.5, 0.15);
+  }
```

Comment:
🔧 I would prefer to pull the statements into separate sections, so it is a bit more clearer.

Like:

```
// given

// when
final var result = testCase.run();

// then
restult.isWithinDeviation(3.5, 0.15);
```